assets/clp_presto/main.py

```
@@ -0,0 +1,114 @@
+        Ingests the dataset at self.datasets_path
+        """
+        #os.system(f"mkdir -p {CLP_PRESTO_HOST_STORAGE}/var/data/baker21")
+        os.system(f'{CLP_PRESTO_HOST_STORAGE}/sbin/compress.sh --timestamp-key {self.timestamp} {self.datasets_path_in_host}')
+        self.sql_execute(f"UPDATE clp_datasets SET archive_storage_directory=\"{CLP_PRESTO_CONTAINER_STORAGE}/var/data/archives/default\" WHERE name=\"default\"")
+    
+    def search(self, query):
+        """
+        Searches an already-ingested dataset with query, which is populated within config.yaml
+        """
+        #return (self.presto_execute(f"USE default; SELECT * from default WHERE {query.strip()[1:-1]}").strip().count('\n') + 1)
+        res = self.presto_execute(f"SELECT msg, c, s, t[1], ctx, id, CAST(attr AS JSON), tags from default WHERE {query.strip()[1:-1]}").strip()
```

Ruff (0.12.2)

Possible SQL injection vector through string-based query construction

(S608)